[Finishes #116027121] Ensure reservation validation

andela-joyebanji opened the pull request:
#### What does this PR do?
- Ensures that inputs (date and duration) supplied when booking a table are validated both on the client side and the server before adding to the cart
- Ensures tables can only be booked 30mins before time
  #### Description of Task to be completed?
- Implement client-side validation of date and duration input
- Implement server-side input validation of date and duration input
#### How should this be manually tested?

> Sign-in as a Diner > Choose table > Try to add the table to cart with an invalid date and time
> 
> Sign-in as a Diner > Choose table > Try to add the table to cart with a date and time less than 30mins from the current time
> 
> Try sending a POST request to _site-root/booking/table/{table_id}/add_ using Postman with incorrect duration and date 
#### What are the relevant pivotal tracker stories?

The corresponding story on _PT_ can be viewed [here](https://www.pivotaltracker.com/story/show/116027121)
#### Screenshots

![Invalid date](https://cloud.githubusercontent.com/assets/17027572/13956779/4932cdc6-f04a-11e5-83a6-a3655fe15ca9.png)
![Incorret date range](https://cloud.githubusercontent.com/assets/17027572/13956780/49515a98-f04a-11e5-9182-8a85baaf9bde.png)

Commit:
[Feature #116027121] Ensure reservation validation

Add book_table javascript file to handle client-side validation of date input
Include moment and moment-timezone library for client-side datetime manipulation
Include links to book_table, moment and moment-timezone file in book.blade.php
Fix wrong alertify method call for error messages
Implement server-side validation for sent table reservation in addTable method in BookingController class

Commit:
Applied fixes from StyleCI

Commit:
Merge pull request #165 from andela/analysis-q1bOW1

[Chore] Applied fixes from StyleCI

andela-anandaa commented:
@andela-joyebanji -- follow the new PR convention, you remember the template?

andela-anandaa commented on app/Http/Controllers/BookingController.php:
```diff
@@ -199,22 +199,54 @@ public function addTable(Request $request)
 
             return redirect('/auth/login');
         }
+       // var_dump($request->date);exit;
```
This should be removed, as opposed to commenting out.

andela-anandaa commented on app/Http/Controllers/BookingController.php:
```diff
@@ -199,22 +199,54 @@ public function addTable(Request $request)
-        ]);
 
-        Session::flash('success', 'Table added to cart');
+        //User can only book 30mins from time. A 2mins delay time is substracted in other to cater for any delay in tranferring the request to the server.
```
This line is too long.

andela-anandaa commented on app/Http/Controllers/BookingController.php:
```diff
@@ -199,22 +199,54 @@ public function addTable(Request $request)
+        $interval = $allowedBookingDateTime->diff($bookingDateTime);
+
+        // Calculate the seconds difference between the allowedBookingDateTime and bookingDateTime
+        //Reference: http://stackoverflow.com/questions/14277611/convert-dateinterval-object-to-seconds-in-php
```
Leave a space after the beginning `//`

andela-anandaa commented on public/js/book_table.js:
```diff
@@ -0,0 +1,39 @@
+      var msg = "";
+      //Parse the date from the form book date form field
+      var bookDate = moment($('#bookDate').val(),"DD/MM/YYYY H:mm", true);
+      if(!bookDate.isValid()){
```
JS convention suggest spaces after `if` and `)`, i.e. `if (...) {`

andela-anandaa commented on public/js/book_table.js:
```diff
@@ -0,0 +1,39 @@
+  }
+
+  $('.fancybox').fancybox();
+  $('form').submit(function(){
```
Same to this, see comment on line 14.

andela-anandaa commented on public/js/book_table.js:
```diff
@@ -0,0 +1,39 @@
+      var next30MinsFromNow = moment().add(30, 'minutes').seconds(0);
+
+      var maxDate = moment.max(bookDate,next30MinsFromNow);
+      if(maxDate.diff(bookDate, 'minutes') > 0){
```
Same here -- spacing.

Commit:
[Chore] Fix notes on PR to staging

Remove commented line in BookingController
Fix comments spacing issues in BookController
Fix spacing issues in book_table

Commit:
[Chore] Merge branch 'ft-ensure-reservations-are-validated-116027121' of https://github.com/andela/resly into ft-ensure-reservations-are-validated-116027121

Commit:
Applied fixes from StyleCI

Commit:
Merge pull request #169 from andela/analysis-qx0Yw0

[Chore] Applied fixes from StyleCI
